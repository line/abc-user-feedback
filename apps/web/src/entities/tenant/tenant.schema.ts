/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import { z } from 'zod';

export const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z
    .array(z.string().refine((v) => /[a-z]+\.[a-z]{2,3}/.test(v)))
    .nullable(),
  useEmailVerification: z.boolean(),
  oauthConfig: z
    .object({
      clientId: z.string(),
      emailKey: z.string(),
      scopeString: z.string(),
      clientSecret: z.string(),
      authCodeRequestURL: z.string(),
      accessTokenRequestURL: z.string(),
      userProfileRequestURL: z.string(),
    })
    .nullable(),
});

export const tenantInfoSchema = tenantSchema.pick({
  id: true,
  siteName: true,
  description: true,
});
export const authInfoScema = tenantSchema.pick({
  isPrivate: true,
  isRestrictDomain: true,
  allowDomains: true,
  useOAuth: true,
  useEmail: true,
  oauthConfig: true,
});
