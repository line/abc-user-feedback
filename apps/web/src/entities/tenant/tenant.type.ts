/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { z } from 'zod';

const oauthConfigSchema = z.object({
  oauthUse: z.boolean(),
  clientId: z.string(),
  clientSecret: z.string(),
  authCodeRequestURL: z.string(),
  scopeString: z.string(),
  accessTokenRequestURL: z.string(),
  userProfileRequestURL: z.string(),
  emailKey: z.string(),
});

const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z.array(z.string()),
  useEmailVerification: z.boolean(),
  oauthConfig: oauthConfigSchema.nullable(),
});

export type Tenant = z.infer<typeof tenantSchema>;

export type OAuthConfig = z.infer<typeof oauthConfigSchema>;
