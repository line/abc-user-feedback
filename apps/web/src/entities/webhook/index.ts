export * from './ui';
export * from './webhook.type';
