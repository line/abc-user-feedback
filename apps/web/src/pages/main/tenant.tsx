/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import type { GetStaticProps } from 'next';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import { parseAsString, useQueryState } from 'nuqs';

import { Icon, Menu, MenuItem } from '@ufb/react';

import type { NextPageWithLayout } from '@/shared';
import SideMenuLayout from '@/shared/ui/side-menu-layout.ui';
import { Layout } from '@/widgets/layout';
import {
  LoginSetting,
  TenantInfoSetting,
  UserManagementSetting,
} from '@/widgets/setting-menu';

import serverSideTranslations from '@/server-side-translations';

const TenantPage: NextPageWithLayout = () => {
  const { t } = useTranslation();
  const router = useRouter();
  const [currentMenu] = useQueryState<string>(
    'menu',
    parseAsString.withDefault('tenant'),
  );

  return (
    <SideMenuLayout
      sideMenu={
        <Menu
          type="single"
          orientation="vertical"
          className="w-full p-0"
          value={currentMenu}
          onValueChange={(value) =>
            router.push({ pathname: router.pathname, query: { menu: value } })
          }
        >
          <MenuItem value="tenant">
            <Icon name="RiInformation2Line" />
            {t('tenant-setting-menu.tenant-info')}
          </MenuItem>
          <MenuItem value="login">
            <Icon name="RiUser2Line" />
            {t('tenant-setting-menu.sign-up-mgmt')}
          </MenuItem>
          <MenuItem value="user">
            <Icon name="RiShieldKeyholeLine" />
            {t('tenant-setting-menu.user-mgmt')}
          </MenuItem>
        </Menu>
      }
    >
      {currentMenu === 'tenant' && <TenantInfoSetting />}
      {currentMenu === 'login' && <LoginSetting />}
      {currentMenu === 'user' && <UserManagementSetting />}
    </SideMenuLayout>
  );
};

TenantPage.getLayout = (page) => {
  return <Layout title="Tenant">{page}</Layout>;
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale)),
    },
  };
};

export default TenantPage;
