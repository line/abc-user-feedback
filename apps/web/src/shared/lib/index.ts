/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
export { default as useOAIQuery } from './useOAIQuery';
export { default as useOAIMutation } from './useOAIMutation';

export { default as useQueryParamsState } from './use-query-params-state';

export { default as useSort } from './use-sort';
export { default as useLocalColumnSetting } from './use-local-column-setting';
export { default as usePermissions } from './use-permissions';

export { default as useHorizontalScroll } from './use-horizontal-scroll';

export { default as sessionStorage } from './session-storage';
export { default as client } from './client';
