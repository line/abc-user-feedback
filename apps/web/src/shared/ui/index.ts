/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
export { default as Logo } from './logo.ui';
export { default as LocaleSelectBox } from './locale-select-box.ui';
export { default as LogoWithTitle } from './logo-with-title.ui';

export { default as SectionTemplate } from './section-template.ui';
export { default as MainCard } from './main-card.ui';

export { default as SubMenu } from './sub-menu.ui';
export { default as CreateTemplate } from './create-template.ui';
export { default as CreateInputTemplate } from './create-input-template.ui';
export { default as CreateSectionTemplate } from './create-section-template.ui.tsx';
export { default as DashboardCard } from './dashboard-card';

export { default as ExpandableText } from './expandable-text.ui';
export { default as ImagePreviewButton } from './image-preview-button';
export { default as ShareButton } from './share-button';
export { default as DateRangePicker } from './date-range-picker';
export { default as DescriptionTooltip } from './description-tooltip';
export { default as HelpCardDocs } from './help-card-docs';
export { default as Popper } from './popper.ui';
export { default as ImageSlider } from './image-slider.ui';
export { default as RadioGroup } from './radio-group';

export { default as SmallCard } from './small-card.ui';

export * from './tables';
export * from './charts';
export * from './select-box';
