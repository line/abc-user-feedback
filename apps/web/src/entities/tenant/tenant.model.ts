/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { create } from 'zustand';

import { client } from '@/shared';

import type { Tenant } from './tenant.type';

type State = { tenant: Tenant | null };

type Action = {
  setTenant: (tenant: Tenant) => void;
  refetchTenant: () => Promise<void>;
};

export const useTenantStore = create<State & Action>((set) => ({
  tenant: null,
  setTenant: (tenant) => set({ tenant }),
  refetchTenant: async () => {
    const { data } = await client.get({ path: '/api/admin/tenants' });
    set({ tenant: data });
  },
}));
