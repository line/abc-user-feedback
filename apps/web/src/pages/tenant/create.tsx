/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import type { GetStaticProps } from 'next';
import { useRouter } from 'next/router';
import { zodResolver } from '@hookform/resolvers/zod';
import { useTranslation } from 'next-i18next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import { useForm } from 'react-hook-form';
import { z } from 'zod';

import { toast } from '@ufb/ui';

import AuthTemplate from '@/components/templates/AuthTemplate';
import { DEFAULT_LOCALE } from '@/constants/i18n';
import { Path } from '@/constants/path';
import { useTenant } from '@/contexts/tenant.context';
import { useOAIMutation } from '@/hooks';
import type { NextPageWithLayout } from '../_app';

interface IForm {
  siteName: string;
}

const schema: Zod.ZodType<IForm> = z.object({
  siteName: z.string(),
});

const defaultValues: IForm = {
  siteName: '',
};

const CreatePage: NextPageWithLayout = () => {
  const { t } = useTranslation();

  const router = useRouter();
  const { refetch } = useTenant();

  const { register, handleSubmit } = useForm({
    resolver: zodResolver(schema),
    defaultValues,
  });

  const { mutate, isPending } = useOAIMutation({
    method: 'post',
    path: '/api/admin/tenants',
    queryOptions: {
      async onSuccess() {
        toast.positive({ title: 'Success' });
        toast.positive({
          title: 'create Default Super User',
          description: 'email: user@feedback.com \n password: 12345678',
        });
        router.push(Path.SIGN_IN);
        refetch();
      },
      onError(error) {
        toast.negative({ title: error?.message ?? 'Error' });
      },
    },
  });
  const onSubmit = (data: IForm) => mutate(data);

  return (
    <AuthTemplate>
      <div className="flex h-screen w-screen items-center justify-center">
        <div className="w-full max-w-[400px] rounded border p-4 shadow">
          <form
            onSubmit={handleSubmit(onSubmit)}
            className="flex flex-col gap-4"
          >
            <h1 className="font-20-bold">{t('tenant.create.title')}</h1>
            <label>
              <span>{t('tenant.create.site-name')}</span>
              <input className="input" type="text" {...register('siteName')} />
            </label>

            <button
              className="btn btn-primary"
              type="submit"
              disabled={isPending}
            >
              {t('button.setting')}
            </button>
          </form>
        </div>
      </div>
    </AuthTemplate>
  );
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale ?? DEFAULT_LOCALE)),
    },
  };
};

export default CreatePage;
