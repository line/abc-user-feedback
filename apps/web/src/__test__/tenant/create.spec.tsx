/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import mockRouter from 'next-router-mock';

import { Path } from '@/shared';
import type { Tenant } from '@/entities/tenant';
import { useTenantStore } from '@/entities/tenant';

import CreateTenantPage from '@/pages/tenant/create';
import { render, waitFor } from '@/test-utils';

describe('Create Tenant Page', () => {
  test('match snapshot', async () => {
    const { container } = render(<CreateTenantPage />);
    expect(container).toMatchSnapshot();
  });

  test('should route sign-in page when tenant is defined', async () => {
    useTenantStore.setState({ tenant: {} as Tenant });
    const createTenantPage = CreateTenantPage.getLayout!(<CreateTenantPage />);

    render(<>{createTenantPage}</>);
    await waitFor(() =>
      expect(mockRouter).toMatchObject({ pathname: Path.SIGN_IN }),
    );
  });
});
