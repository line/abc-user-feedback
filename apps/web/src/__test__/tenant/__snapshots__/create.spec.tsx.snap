// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Tenant Page match snapshot 1`] = `
<div>
  <form
    class="flex flex-col gap-4"
  >
    <h1
      class="font-20-bold"
    >
      tenant.create.title
    </h1>
    <label>
      <span>
        tenant.create.site-name
      </span>
      <input
        class="input"
        name="siteName"
        placeholder="Please enter the site name"
        type="text"
      />
    </label>
    <button
      class="btn btn-primary"
      disabled=""
      type="submit"
    >
      button.setting
    </button>
  </form>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;
